pkg/retry/errors.go

```diff
@@ -35,3 +36,15 @@ func IsRetryableError(err error) bool {
 	}
 	return false
 }
+
+// IsConnectionError tells whether this error should reconnect to Database
+func IsConnectionError(err error) bool {
+	err = errors.Cause(err)
+	switch err {
+	case driver.ErrBadConn:
+		return true
+	case mysql.ErrInvalidConn:
+		return true
```

Comment:
can we retry for `mysql.ErrInvalidConn` without `safe-mode`?

Reply:
yes, we should better not retry, I'll remove it

Reply:
remove it [here](https://github.com/pingcap/dm/pull/266/files#diff-17b57264f15f5001c686d1232127256fR44)